### PR TITLE
New user welcome email

### PR DESCRIPTION
**Pull request by fillup**

### CODE REVIEW

File: application/listeners/listeners.go [hunk 1]
```diff
@@ -172,3 +186,23 @@ func sendPostCreatedNotifications(e events.Event) {
...
 }
+
+func sendNewUserWelcome(user models.User) error {
+	if user.Email == "" {
+		return errors.New("'To' email address is required")
+	}
+
+	msg := notifications.Message{
+		Template:  domain.MessageTemplateNewUserWelcome,
+		ToName:    user.FirstName + " " + user.LastName,
```
Comment (Schparky) [inline application/listeners/listeners.go]:
```suggestion
		ToName:    user.GetRealName(),
```

File: application/templates/mail/new_user_welcome.plush.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<h3>{{.firstName}} - Welcome to {{.appName}}!</h3>
+
+<p>
+    We (the team behind {{.appName}}) appreciate you coming to check us out. We're excited about the potential for {{.appName}}
```
Comment (Schparky) [inline application/templates/mail/new_user_welcome.plush.html]:
```suggestion
    We (the team behind {{.appName}}) appreciate you coming to check out {{.appName}. We're excited about its potential
```

File: application/listeners/listeners.go [hunk 3]
```diff
@@ -101,7 +103,19 @@ func userCreated(e events.Event) {
 		return
 	}
 
+	user, ok := e.Payload["user"].(*models.User)
+	if !ok {
+		domain.Logger.Printf("%s Failed to get User from event payload for notification. Event message: %s",
+			domain.GetCurrentTime(), e.Message)
```
Comment (Schparky) [inline application/listeners/listeners.go]: I haven't been adding the time to log messages, do you think I should? FWIW, CloudWatch logs the time too. Is that redundant?

Comment (fillup) [reply application/listeners/listeners.go]: okay, yeah I'd say it's redundant and we can remove it. perhaps we had it there for stdout to our console, but that we'd be watching realtime during dev/test anyway

File: application/listeners/listeners.go [hunk 4]
```diff
@@ -172,3 +185,23 @@ func sendPostCreatedNotifications(e events.Event) {
...
+		return errors.New("'To' email address is required")
+	}
+
+	msg := notifications.Message{
+		Template:  domain.MessageTemplateNewUserWelcome,
+		ToName:    user.GetRealName(),
+		ToEmail:   user.Email,
+		FromEmail: domain.Env.EmailFromAddress,
+		Data: map[string]interface{}{
+			"appName":      domain.Env.AppName,
```
Comment (Baggerone) [inline application/listeners/listeners.go]: Looks like your email template is expecting firstName as well.

File: application/templates/mail/new_user_welcome.plush.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        <li>{{.appName}} URL: {{.uiURL}}</li>
+        <li>Your email address for login: {{.userEmail}}</li>
+        <li>Support: {{.supportEmail}}</li>
+    </ul>
+</p>
+
+<h3>Tell your friends about {{.appName}}!</h3>
+<p>
+    If you have friends in other organizations that you think could benefit from {{.appName}}, have them visit
+    <a href="https://www.wecarry.app">https://www.wecarry.app</a> or contact us at
```
[the remark this answers is not in the document]
Comment (fillup) [reply application/templates/mail/new_user_welcome.plush.html]: I thought about it, but we won't have multiple marketing site environments, so I think it is fine.

File: application/listeners/listeners.go (hunk 4 above)
Comment (fillup) [reply application/listeners/listeners.go]: great catch, thanks!

File: application/templates/mail/new_user_welcome.plush.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (Baggerone) [reply application/templates/mail/new_user_welcome.plush.html]: Even though this is open source?

**Pull request merged**